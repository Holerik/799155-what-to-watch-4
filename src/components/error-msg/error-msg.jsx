// error-msg.jsx
import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getErrMessage} from '../../reducer/error/selectors.js';

const ErrorMsg = React.memo(function ErrorMsg(props) {
  const {errMessage} = props;
  return (
    <div className="error-message">
      <p>{errMessage}</p>
    </div>
  );
});

const mapStateToProps = (state) => ({
  errMessage: getErrMessage(state),
});

ErrorMsg.propTypes = {
  errMessage: PropTypes.string.isRequired,
};

export {ErrorMsg};
export default connect(mapStateToProps, null)(ErrorMsg);

