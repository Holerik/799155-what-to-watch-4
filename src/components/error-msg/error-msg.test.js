// error-msg.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {ErrorMsg} from './error-msg.jsx';

const mockStore = configureStore([]);

describe(`ErrorMsg test`, () => {
  it(`ErrorMsg component should render correctly`, () => {
    const store = mockStore({
      ERROR: {
        message: ``,
        show: false,
      },
    });
    const tree = renderer
    .create(
        <Provider store={store}>
          <ErrorMsg
            errMessage={`Error 401. Unauthorized status`}
          />)
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

