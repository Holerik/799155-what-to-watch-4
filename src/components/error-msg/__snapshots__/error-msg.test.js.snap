// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMsg test ErrorMsg component should render correctly 1`] = `
Array [
  <div
    className="error-message"
  >
    <p>
      Error 401. Unauthorized status
    </p>
  </div>,
  ")",
]
`;
